src/anndata/_core/merge.py

```diff
@@ -939,21 +939,34 @@ def gen_outer_reindexers(els, shapes, new_index: pd.Index, *, axis=0):
     return reindexers
 
 
+def missing_element(
+    n: int,
+    els: list[SpArray | sparse.csr_matrix | sparse.csc_matrix | np.ndarray | DaskArray],
+    axis: Literal[0, 1] = 0,
+    fill_value: Any | None = None,
+) -> np.ndarray | DaskArray:
+    """Generates value to use when there is a missing element."""
+    should_return_dask = any(isinstance(el, DaskArray) for el in els)
+    non_missing_elem = next(el for el in els if not_missing(el))
```

Comment:
This will raise a `StopIteration` when calling `missing_element(n, [])`.

Maybe you should catch it and transform it into a `ValueError`?

Reply:
> StopIteration might leak out into a loop higher in the stack and cancel it.

What does this mean?  Would't raising a ValueError also do this?

Reply:
To understand what I mean, you need to understand how a `for` loop works in Python.

It’s basically syntactic sugar around `iter()`, `next()`, and `StopIteration`:

```py
for <var> in <expr>:
    <block>
```

desugars to:

```py
_it = iter(<expr>)
while True:
    try:
        var = next(_it)
    except StopIteration:
        break
    <block>
```